### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by wkliao** [Parallel-NetCDF/E3SM-IO#96]
https://github.com/Parallel-NetCDF/E3SM-IO/blob/6955e3f76eb4ef332c979a8371f2b7bd9e3955f0/src/drivers/e3sm_io_driver_hdf5_agg.cpp#L246

```
e3sm_io_driver_hdf5_agg.cpp:246:73: error: ‘local_no_collective_cause’ was not declared in this scope
         H5Pget_mpio_no_collective_cause (this->driver.dxplid_coll, &local_no_collective_cause,
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~

e3sm_io_driver_hdf5_agg.cpp:308:73: error: ‘local_no_collective_cause’ was not declared in this scope
         H5Pget_mpio_no_collective_cause (this->driver.dxplid_coll, &local_no_collective_cause,
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by wkliao** [comment]
Does 094d9ad675844306e8f0a0f80370cdeb452af2a7 fix this?

**Comment by khou2020** [comment]
It should be fixed long ago.

**Comment by wkliao** [comment]
You can give it a try to build with a regular HDF5 library without multi-dataset.

**Comment by khou2020** [comment]
> You can give it a try to build with a regular HDF5 library without multi-dataset.
In previous meetings, we decided to drop support to non-multi-dataset HDF5 due to performance issues.

**Comment by wkliao** [comment]
This is compile error, not options currently supported or not.

**Comment by khou2020** [comment]
> This is compile error, not options currently supported or not.

It did not happen on my side.
The section should not be compiled when multi-dataset was not found.

**Comment by wkliao** [comment]
Did you try the version before or after 094d9ad?

**Comment by khou2020** [comment]
> Did you try the version before or after [094d9ad](https://github.com/Parallel-NetCDF/E3SM-IO/commit/094d9ad675844306e8f0a0f80370cdeb452af2a7)?

I reorganized the code in another PR. That should solve all the problems.
I'll post it after the current PR.

**Comment by wkliao** [comment]
fixed in #98
